instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java

```diff
@@ -0,0 +1,283 @@
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepStart.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepLeft")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepLeft.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(3))));
+  }
+
+  @Test
+  void shouldTracePartitionedJob() {
+    runner.runJob("partitionedJob");
+
+    testing.waitAndAssertTraces(
+        trace -> {
+          int index = 2;
+          trace.hasSpansSatisfyingExactly(
+              span ->
+                  span.hasName("BatchJob partitionedJob")
+                      .hasKind(SpanKind.INTERNAL)
+                      .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+              span ->
+                  span.hasName(
+                          hasPartitionManagerStep()
+                              ? "BatchJob partitionedJob.partitionManagerStep"
+                              : "BatchJob partitionedJob.partitionWorkerStep")
+                      .hasKind(SpanKind.INTERNAL)
+                      .hasParent(trace.getSpan(0)),
+              span ->
+                  span.satisfies(
+                          spanData ->
+                              assertThat(spanData.getName())
+                                  .matches(
+                                      "BatchJob partitionedJob.partitionWorkerStep:partition[01]"))
+                      .hasKind(SpanKind.INTERNAL)
+                      .hasParent(trace.getSpan(1)),
+              span -> partitionChunk(trace, span, index),
+              span -> partitionChunk(trace, span, index),
+              span ->
+                  span.satisfies(
+                          spanData ->
+                              assertThat(spanData.getName())
+                                  .matches(
+                                      "BatchJob partitionedJob.partitionWorkerStep:partition[01]"))
+                      .hasParent(trace.getSpan(1)),
+              span -> partitionChunk(trace, span, 5),
+              span -> partitionChunk(trace, span, 5));
+        });
+  }
+
+  private static void partitionChunk(TraceAssert trace, SpanDataAssert span, int index) {
+    span.satisfies(
+            spanData ->
+                assertThat(spanData.getName())
+                    .matches("BatchJob partitionedJob.partitionWorkerStep:partition[01].Chunk"))
+        .hasParent(trace.getSpan(index));
+  }
+
+  protected boolean hasPartitionManagerStep() {
+    return true;
+  }
+
+  // should be moved to SpanDataAssert
```

Comment:
what does this comment mean?

Reply:
oh - just saw that this method already exists: `hasException`